lib/web_ui/lib/src/engine/js_interop/js_promise.dart

```diff
@@ -6,18 +6,39 @@
 library js_promise;
 
 import 'package:js/js.dart';
+import 'package:js/js_util.dart' as js_util;
+
+@JS()
+@staticInterop
+class PromiseResolver<T extends Object> {}
+
+extension PromiseResolverExtension<T extends Object> on PromiseResolver<T> {
+  void resolve(T result) => js_util.callMethod(this, 'call', <Object>[this, result]);
+}
+
+@JS()
+@staticInterop
+class PromiseRejecter {}
+
+extension PromiseRejecterExtension on PromiseRejecter {
+  void reject(Object? error) => js_util.callMethod(this, 'call', <Object>[this, if (error != null) error]);
+}
 
 /// Type-safe JS Promises
 @JS('Promise')
 @staticInterop
-abstract class Promise<T> {
+abstract class Promise<T extends Object> {
```

Comment:
js-promise values can never be null?

Reply:
Hmmm... I think I can enable nullability here, and we'll interpret dart null as undefined (not passing the argument at all to `resolve`)

Reply:
IIUC `extends Object` only ever really means "this will never be `null`"

Reply:
Well we don't have any Promise objects currently that have a nullable object. And how nullability maps to JS from wasm is kind of fuzzy right now, sometimes it means JS null, sometimes it means undefined. So I was kind of trying to avoid it, but I think allowing it to be nullable and using undefined here is probably okay.